### PR TITLE
Google Analytics data source date parameter enhancement

### PR DESCRIPTION
**Pull request by hidekoji**
# Description

This PR enhances GA's Date Parameters.

# Checklist
Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [ ] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/google_analytics.R [hunk 1]
```diff
@@ -31,12 +44,28 @@ getGoogleAnalytics <- function(tableId, lastNDays = 30, dimensions, metrics, tok
   }
   token <- getGoogleTokenForAnalytics(tokenFileId)
 
-  start_date <- as.character(lubridate::today() - lubridate::days(lastNDays))
-  #end_date <- as.character(lubridate::today() - lubridate::days(1))
-  end_date <- as.character(lubridate::today())
+  if(dateRangeType == "lastNDays") {
+    if(is.null(lastN)) {
+      lastN <- lastNDays # For backward compatibility
```
Comment (hideaki) [inline R/google_analytics.R]: Please leave comment on detail of the backward compatibility this is about.

Comment (hidekoji) [reply R/google_analytics.R]: Added commentl

File: R/google_analytics.R [hunk 2]
```diff
@@ -19,7 +19,20 @@ getGoogleAnayticsSegmentList <- function(){
 }
 
 #' @export
-getGoogleAnalytics <- function(tableId, lastNDays = 30, dimensions, metrics, tokenFileId = NULL, paginate_query=FALSE, segments = NULL, ...){
+#' @param tableId - GA's table id (a.k.a viewID).
+#' @param lastNDays - Deprecated. Use dateRangeType and lastN arguments instead.
+#' @param dimensions - GA's dimensions
+#' @param metrics - GA's metrics.
+#' @param tokenFileId - Optional. for GA data source created with old Exploratory Desktop.
+#' @param pagenamte_query - for pagination
```
Comment (hideaki) [inline R/google_analytics.R]: Typo: pagenamte_query

File: R/google_analytics.R [hunk 3]
```diff
@@ -31,12 +44,31 @@ getGoogleAnalytics <- function(tableId, lastNDays = 30, dimensions, metrics, tok
   }
   token <- getGoogleTokenForAnalytics(tokenFileId)
 
-  start_date <- as.character(lubridate::today() - lubridate::days(lastNDays))
-  #end_date <- as.character(lubridate::today() - lubridate::days(1))
-  end_date <- as.character(lubridate::today())
+  if(dateRangeType == "lastNDays") {
+    if(is.null(lastN)) {
+      # For backward compatibility for Exploratory Desktop older than verssion 5.4.1
```
Comment (hideaki) [inline R/google_analytics.R]: Typo: verssion

File: R/google_analytics.R (hunk 2 above)
Comment (hidekoji) [reply R/google_analytics.R]: Fixed it as paginate_query.

File: R/google_analytics.R (hunk 3 above)
Comment (hidekoji) [reply R/google_analytics.R]: Thank you for catching it. Fixed the typo.

File: R/google_analytics.R [hunk 4]
```diff
@@ -19,7 +19,20 @@ getGoogleAnayticsSegmentList <- function(){
 }
 
 #' @export
-getGoogleAnalytics <- function(tableId, lastNDays = 30, dimensions, metrics, tokenFileId = NULL, paginate_query=FALSE, segments = NULL, ...){
+#' @param tableId - GA's table id (a.k.a viewID).
+#' @param lastNDays - Deprecated. Use dateRangeType and lastN arguments instead.
+#' @param dimensions - GA's dimensions
+#' @param metrics - GA's metrics.
+#' @param tokenFileId - Optional. for GA data source created with old Exploratory Desktop.
+#' @param paginamte_query - for pagination
```
Comment (hideaki) [inline R/google_analytics.R]:
```suggestion
#' @param paginate_query - for pagination
```

**Pull request merged**